snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/rule664.groovy

```diff
@@ -92,12 +92,13 @@ if (params.isUnpublishedOnly) {
 		.each { Hits<String[]> publishedTermsBatch ->
 			// all returned terms are duplicate of an unpublished term
 			publishedTermsBatch.each { publishedTerm ->
-				if (activeConceptIds.get().contains(publishedTerm[1])) {
-					issues.add(ComponentIdentifier.of(SnomedDescription.TYPE, publishedTerm[0]))
-				}
+				String term = publishedTerm[0];
+				descriptionsByTerm.get(term).forEach({ descriptionId -> 
+					//Report the unpublished term
+					issues.add(ComponentIdentifier.of(SnomedDescription.TYPE,  descriptionId))
```

Comment:
Also "unpublished vs. unpublished" cases get two validation issues, whereas "unpublished vs. published" might only get one. You could also report the description ID that is returned by the query &ndash; except it is no longer part of the selected fields.

Reply:
Modified this to report both in case of unpublished only runs